topology.go

```diff
@@ -156,7 +164,51 @@ func QueryTopology(dbpool *pgxpool.Pool, params TopologyOptions) ([]*models.Comp
 	// If ID is present, we do not apply any filters to the root component
 	results = applyStatusFilter(results, params.ID != "", params.Status...)
 
-	return results, nil
+	var res TopologyResponse
+	res.Components = results
+	populateTopologyResult(results, &res)
+
+	res.Teams, err = GetTeamNamesOfComponents(ctx, dbpool, res.componentIDs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get teams of components: %w", err)
+	}
+
+	return &res, nil
+}
+
+func GetTeamNamesOfComponents(ctx context.Context, dbpool *pgxpool.Pool, componentIDs []string) ([]string, error) {
+	if len(componentIDs) == 0 {
```

Comment:
Can we not just update the views to return teams per component?

Reply:
We're not using the `topology` view right now. I'll leave this as it is for now since we'll be migrating to views anyway.